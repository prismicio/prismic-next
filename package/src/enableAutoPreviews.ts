import { PreviewData } from "next";
import { Client, HttpRequestLike } from "@prismicio/client";
import { NextContextLike } from "./types";

interface PrismicNextPreviewData {
	ref: string;
}

const isPrismicNextPreviewData = (
	previewData: PreviewData,
): previewData is PrismicNextPreviewData =>
	typeof previewData === "object" && "ref" in previewData;

export type EnableAutoPreviewsConfig<
	TPreviewData extends PreviewData = PreviewData,
> = {
	client: Client;
} & (
	| {
			context?: NextContextLike<TPreviewData>;
	  }
	| {
			req?: HttpRequestLike;
	  }
);
/**
 * Configures a Prismic client to automatically query draft content during a
 * preview session. Either takes in a Next.js `getStaticProps` context object or
 * a Next.js API endpoint request object.
 *
 * @param config - Configuration for the function.
 */
export const enableAutoPreviews = <
	TPreviewData extends PreviewData = PreviewData,
>(
	config: EnableAutoPreviewsConfig<TPreviewData>,
): void => {
	/**
	 * If preview data is being passed from Next Context then use queryContentFromRef
	 */
	if ("context" in config && config.context) {
		const previewData = config.context.previewData;

		if (isPrismicNextPreviewData(previewData) && previewData.ref) {
			config.client.queryContentFromRef(previewData.ref);
		}
		/**
		 * If the req object is passed then use enableAutoPreviewsFromReq
		 */
	} else if ("req" in config && config.req) {
		config.client.enableAutoPreviewsFromReq(config.req);
	}
};
