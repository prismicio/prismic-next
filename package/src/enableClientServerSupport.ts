import { GetStaticPropsContext, NextApiRequest, PreviewData } from "next";
import * as prismic from "@prismicio/client";
import { NextContextLike } from "./types";

interface PrismicNextPreviewData {
	ref: string;
}

const isPrismicNextPreviewData = (
	previewData: PreviewData,
): previewData is PrismicNextPreviewData =>
	typeof previewData === "object" && "ref" in previewData;

type EnableClientServerSupportConfig = {
	client: prismic.Client;
} & (
	| {
			context?: NextContextLike;
	  }
	| {
			req?: NextApiRequest;
	  }
);

export const enableClientServerSupport = (
	config: EnableClientServerSupportConfig,
): void => {
	if ("context" in config && config.context) {
		const previewData = config.context.previewData;

		if (isPrismicNextPreviewData(previewData) && previewData.ref) {
			config.client.queryContentFromRef(previewData.ref);
		}
	} else if ("req" in config && config.req) {
		config.client.enableAutoPreviewsFromReq(config.req);
	}
};
