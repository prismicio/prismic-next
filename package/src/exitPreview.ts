import { NextApiResponse, NextApiRequest } from "next";

/**
 * Configuration for `exitPreview`.
 */
export type ExitPreviewConfig = {
	/**
	 * The `req` object from a Next.js API route. This is given as a parameter to
	 * the API route.
	 *
	 * @see Next.js API route docs: {@link https://nextjs.org/docs/api-routes/introduction}
	 */
	req: {
		headers: {
			referer?: NextApiRequest["headers"]["referer"];
		};
	};

	/**
	 * The `res` object from a Next.js API route. This is given as a parameter to
	 * the API route.
	 *
	 * @see Next.js API route docs: {@link https://nextjs.org/docs/api-routes/introduction}
	 */
	res: {
		clearPreviewData: NextApiResponse["clearPreviewData"];
		redirect: NextApiResponse["redirect"];
	};
};

/**
 * Exits Next.js's Preview Mode from within a Next.js API route.
 *
 * If the user was sent to the endpoint from a page, the user will be redirected
 * back to that page after exiting Preview Mode.
 */
export function exitPreview(config: ExitPreviewConfig): void {
	const { req } = config;
	// Exit the current user from "Preview Mode". This function accepts no args.
	config.res.clearPreviewData();

	if (req.headers.referer) {
		const url = new URL(req.headers.referer);

		if (url.pathname !== "/api/exit-preview") {
			// Redirect the user to the referrer page.
			config.res.redirect(req.headers.referer);

			return;
		}
	}

	config.res.redirect("/");
}
