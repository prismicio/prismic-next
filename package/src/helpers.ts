import { GetStaticPropsContext, PreviewData } from "next";
import * as prismic from "@prismicio/client";

interface PrismicNextPreviewData {
	ref: string;
}

const isPrismicNextPreviewData = (
	previewData: PreviewData,
): previewData is PrismicNextPreviewData =>
	typeof previewData === "object" && "ref" in previewData;

export const getPreviewRefFromContext = (
	context: GetStaticPropsContext,
): string | undefined => {
	if (isPrismicNextPreviewData(context.previewData)) {
		return context.previewData.ref;
	}
};

export const enableClientServerSupport = (
	client: prismic.Client,
	context: GetStaticPropsContext,
): void => {
	const ref = getPreviewRefFromContext(context);

	if (ref) {
		client.queryContentFromRef(ref);
	}
};
