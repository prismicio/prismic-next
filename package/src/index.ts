export type { SetPreviewDataConfig } from "./setPreviewData";
export type { EnableAutoPreviewsConfig } from "./enableAutoPreviews";
export type { ExitPreviewParams } from "./exitPreview";
export { setPreviewData } from "./setPreviewData";
export { exitPreview } from "./exitPreview";
export { PrismicPreview } from "./PrismicPreview";
export { enableAutoPreviews } from "./enableAutoPreviews";
export { redirectToPreviewURL } from "./redirectToPreviewURL";
export type {
	NextContextLike,
	CreateClientConfig,
	PreviewConfig,
} from "./types";
