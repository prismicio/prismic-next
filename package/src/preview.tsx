import React, { useEffect } from "react";

type PrismicPreviewConfig = {
	repoName: string;
	children: React.ReactChild[] | React.ReactChild;
	updatePreviewURL?: string;
	exitPreviewURL?: string;
};

export function PrismicPreview({
	repoName,
	children,
	updatePreviewURL = "/api/preview",
	exitPreviewURL = "/api/exit-preview",
}: PrismicPreviewConfig) {
	useEffect(() => {
		const prismicPreviewUpdate = async (event: Event) => {
			// Prevent the toolbar from reloading the page.
			event.preventDefault();

			const detail = (event as CustomEvent<{ ref: string }>).detail;

			// Update the preview cookie.
			await fetch(`${updatePreviewURL}?token=${detail.ref}`);

			// Reload the page with the updated token.
			window.location.reload();
		};

		const prismicPreviewEnd = async (e: Event) => {
			e.preventDefault();
			await fetch(exitPreviewURL);
			window.location.reload();
		};

		if (window) {
			window.addEventListener("prismicPreviewUpdate", prismicPreviewUpdate);

			window.addEventListener("prismicPreviewEnd", prismicPreviewEnd);
		}

		return () => {
			if (window) {
				window.removeEventListener(
					"prismicPreviewUpdate",
					prismicPreviewUpdate,
				);
				window.removeEventListener("prismicPreviewEnd", prismicPreviewEnd);
			}
		};
	}, []);
	return (
		<React.Fragment>
			<script
				data-prismic-toolbar=""
				data-repository-name={repoName}
				async
				defer
				src={`https://static.cdn.prismic.io/prismic.js?new=true&repo=${repoName}`}
			></script>
			{children}
		</React.Fragment>
	);
}
