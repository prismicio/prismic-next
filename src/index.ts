export type { SetPreviewDataConfig } from "./setPreviewData";
export type { EnableAutoPreviewsConfig } from "./enableAutoPreviews";
export type { ExitPreviewConfig as ExitPreviewParams } from "./exitPreview";
export type { PrismicPreviewProps } from "./PrismicPreview";
export { setPreviewData } from "./setPreviewData";
export { exitPreview } from "./exitPreview";
export { PrismicPreview } from "./PrismicPreview";
export { enableAutoPreviews } from "./enableAutoPreviews";
export { redirectToPreviewURL } from "./redirectToPreviewURL";
export type { CreateClientConfig, PreviewConfig } from "./types";
