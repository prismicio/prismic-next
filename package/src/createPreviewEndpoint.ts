import { NextApiResponse, NextApiRequest } from "next";
import { LinkResolverFunction } from "@prismicio/helpers";
import { Client } from "@prismicio/client";

export type PreviewConfig = {
	req: NextApiRequest;
	res: NextApiResponse;
	client: Client;
	linkResolver: LinkResolverFunction;
};

export async function createPreviewEndpoint({
	req,
	res,
	client,
	linkResolver,
}: PreviewConfig): Promise<void> {
	const { token: ref } = req.query;

	client.enableAutoPreviewsFromReq(req);

	const previewUrl = await client.resolvePreviewURL({
		linkResolver,
		defaultURL: "/",
	});

	res.setPreviewData({ ref });
	await res.redirect(previewUrl);
}
