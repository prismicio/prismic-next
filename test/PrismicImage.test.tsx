import { test, expect, vi } from "vitest";
import * as React from "react";
import * as renderer from "react-test-renderer";
import * as prismicM from "@prismicio/mock";

import { PrismicNextImage } from "../src";

/**
 * Renders a JSON representation of a React.Element. This is a helper to reduce
 * boilerplate in each test.
 *
 * @param element - The React.Element to render.
 *
 * @returns The JSON representation of `element`.
 */
export const renderJSON = (
	element: React.ReactElement,
	options?: renderer.TestRendererOptions,
): renderer.ReactTestRendererJSON | null => {
	let root: renderer.ReactTestRenderer;

	renderer.act(() => {
		root = renderer.create(element, options);
	});

	// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
	return root!.toJSON() as renderer.ReactTestRendererJSON;
};

const getImg = (
	rendererJSON: renderer.ReactTestRendererJSON | null,
): renderer.ReactTestRendererJSON | undefined => {
	const noscript = rendererJSON?.children?.find(
		(child) => (child as renderer.ReactTestRendererJSON).type === "noscript",
	) as renderer.ReactTestRendererJSON;

	if (noscript.children) {
		return noscript.children.find(
			(child) => (child as renderer.ReactTestRendererJSON).type === "img",
		) as renderer.ReactTestRendererJSON;
	}
};

test("renders null when passed an empty field", () => {
	const seed = expect.getState().currentTestName;
	const field = prismicM.value.image({ seed, state: "empty" });

	const actual = renderJSON(<PrismicNextImage field={field} />);

	expect(actual).toBe(null);
});

test("renders a NextImage for a given field", () => {
	const seed = expect.getState().currentTestName;
	const field = prismicM.value.image({ seed });

	const actual = renderJSON(<PrismicNextImage field={field} />);

	expect(actual).toMatchSnapshot();
});

test('supports layout="responsive"', () => {
	const seed = expect.getState().currentTestName;
	const field = prismicM.value.image({ seed });

	const actual = renderJSON(
		<PrismicNextImage field={field} layout="responsive" />,
	);

	expect(actual).toMatchSnapshot();
});

test('supports layout="fill"', () => {
	const seed = expect.getState().currentTestName;
	const field = prismicM.value.image({ seed });

	const actual = renderJSON(<PrismicNextImage field={field} layout="fill" />);

	expect(actual).toMatchSnapshot();
});

test('supports layout="fixed"', () => {
	const seed = expect.getState().currentTestName;
	const field = prismicM.value.image({ seed });

	const actual = renderJSON(<PrismicNextImage field={field} layout="fixed" />);

	expect(actual).toMatchSnapshot();
});

test('supports layout="intrinsic"', () => {
	const seed = expect.getState().currentTestName;
	const field = prismicM.value.image({ seed });

	const actual = renderJSON(
		<PrismicNextImage field={field} layout="intrinsic" />,
	);

	expect(actual).toMatchSnapshot();
});

test("uses the field's alt if given", () => {
	const seed = expect.getState().currentTestName;
	const field = prismicM.value.image({ seed });
	field.alt = "foo";

	const actual = renderJSON(<PrismicNextImage field={field} />);
	const img = getImg(actual);

	expect(img?.props.alt).toBe(field.alt);
});

test("alt is undefined if the field does not have an alt value", () => {
	const seed = expect.getState().currentTestName;
	const field = prismicM.value.image({ seed });
	field.alt = null;

	const actual = renderJSON(<PrismicNextImage field={field} />);
	const img = getImg(actual);

	expect(img?.props.alt).toBe(undefined);
});

test("supports an explicit decorative fallback alt value if given", () => {
	const seed = expect.getState().currentTestName;
	const field = prismicM.value.image({ seed });
	field.alt = null;

	const actual = renderJSON(<PrismicNextImage field={field} fallbackAlt="" />);
	const img = getImg(actual);

	expect(img?.props.alt).toBe("");
});

test("warns if a non-decorative fallback alt value is given", () => {
	const seed = expect.getState().currentTestName;
	const field = prismicM.value.image({ seed });

	const consoleWarnSpy = vi
		.spyOn(console, "warn")
		.mockImplementation(() => void 0);

	renderJSON(
		// @ts-expect-error - Purposely giving incompatible props.
		<PrismicNextImage field={field} fallbackAlt="non-decorative" />,
	);

	expect(consoleWarnSpy).toHaveBeenCalledWith(
		expect.stringMatching(/alt-must-be-an-empty-string/),
	);

	consoleWarnSpy.mockRestore();
});

test("supports an explicit decorative alt when field has an alt value", () => {
	const seed = expect.getState().currentTestName;
	const field = prismicM.value.image({ seed });
	field.alt = "provided alt";

	const actual = renderJSON(<PrismicNextImage field={field} alt="" />);
	const img = getImg(actual);

	expect(img?.props.alt).toBe("");
});

test("supports an explicit decorative alt when field does not have an alt value", () => {
	const seed = expect.getState().currentTestName;
	const field = prismicM.value.image({ seed });
	field.alt = null;

	const actual = renderJSON(<PrismicNextImage field={field} alt="" />);
	const img = getImg(actual);

	expect(img?.props.alt).toBe("");
});

test("warns if a non-decorative alt value is given", () => {
	const seed = expect.getState().currentTestName;
	const field = prismicM.value.image({ seed });

	const consoleWarnSpy = vi
		.spyOn(console, "warn")
		.mockImplementation(() => void 0);

	renderJSON(
		// @ts-expect-error - Purposely giving incompatible props.
		<PrismicNextImage field={field} alt="non-decorative" />,
	);

	expect(consoleWarnSpy).toHaveBeenCalledWith(
		expect.stringMatching(/alt-must-be-an-empty-string/),
	);

	consoleWarnSpy.mockRestore();
});

test("supports imgix parameters", () => {
	const seed = expect.getState().currentTestName;
	const field = prismicM.value.image({ seed });

	const actual = renderJSON(
		<PrismicNextImage field={field} imgixParams={{ sat: -100 }} />,
	);
	const img = getImg(actual);
	const src = new URL(img?.props.src);

	expect(src.searchParams.get("sat")).toBe("-100");
});

test("applies fit=max by default", () => {
	const seed = expect.getState().currentTestName;
	const field = prismicM.value.image({ seed });
	const fieldURL = new URL(field.url);
	fieldURL.searchParams.delete("fit");
	field.url = fieldURL.toString();

	const actual = renderJSON(<PrismicNextImage field={field} />);
	const img = getImg(actual);
	const src = new URL(img?.props.src);

	expect(src.searchParams.get("fit")).toBe("max");
});

test("retains fit parameter if already included in the image url", () => {
	const seed = expect.getState().currentTestName;
	const field = prismicM.value.image({ seed });
	const fieldURL = new URL(field.url);
	fieldURL.searchParams.set("fit", "crop");
	field.url = fieldURL.toString();

	const actual = renderJSON(<PrismicNextImage field={field} />);
	const img = getImg(actual);
	const src = new URL(img?.props.src);

	expect(src.searchParams.get("fit")).toBe("crop");
});

test("allows overriding fit via imgixParams prop", () => {
	const seed = expect.getState().currentTestName;
	const field = prismicM.value.image({ seed });
	const fieldURL = new URL(field.url);
	fieldURL.searchParams.set("fit", "crop");
	field.url = fieldURL.toString();

	const actual = renderJSON(
		<PrismicNextImage field={field} imgixParams={{ fit: "facearea" }} />,
	);
	const img = getImg(actual);
	const src = new URL(img?.props.src);

	expect(src.searchParams.get("fit")).toBe("facearea");
});

test("allows falling back to the default loader", () => {
	const seed = expect.getState().currentTestName;
	const field = prismicM.value.image({ seed });

	const actual = renderJSON(
		<PrismicNextImage field={field} loader={undefined} />,
	);
	const img = getImg(actual);

	expect(img?.props.src).toMatch(/^\/_next\/image\?url=/);
});
