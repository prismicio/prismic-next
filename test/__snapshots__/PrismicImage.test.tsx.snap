// Vitest Snapshot v1

exports[`renders a NextImage for a given field 1`] = `
<span
  style={
    {
      "background": "none",
      "border": 0,
      "boxSizing": "border-box",
      "display": "inline-block",
      "height": "initial",
      "margin": 0,
      "maxWidth": "100%",
      "opacity": 1,
      "overflow": "hidden",
      "padding": 0,
      "position": "relative",
      "width": "initial",
    }
  }
>
  <span
    style={
      {
        "background": "none",
        "border": 0,
        "boxSizing": "border-box",
        "display": "block",
        "height": "initial",
        "margin": 0,
        "maxWidth": "100%",
        "opacity": 1,
        "padding": 0,
        "width": "initial",
      }
    }
  >
    <img
      alt=""
      aria-hidden={true}
      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%276000%27%20height=%274000%27/%3e"
      style={
        {
          "background": "none",
          "border": 0,
          "display": "block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "padding": 0,
          "width": "initial",
        }
      }
    />
  </span>
  <img
    alt="In hendrerit gravida rutrum quisque non tellus orci ac auctor augue mauris augue"
    data-nimg="intrinsic"
    decoding="async"
    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
    style={
      {
        "border": "none",
        "bottom": 0,
        "boxSizing": "border-box",
        "display": "block",
        "height": 0,
        "left": 0,
        "margin": "auto",
        "maxHeight": "100%",
        "maxWidth": "100%",
        "minHeight": "100%",
        "minWidth": "100%",
        "objectFit": undefined,
        "objectPosition": undefined,
        "padding": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 0,
      }
    }
  />
  <noscript>
    <img
      alt="In hendrerit gravida rutrum quisque non tellus orci ac auctor augue mauris augue"
      data-nimg="intrinsic"
      decoding="async"
      loading="lazy"
      src="https://images.unsplash.com/photo-1446329813274-7c9036bd9a1f?w=3840&fit=crop"
      srcSet="https://images.unsplash.com/photo-1446329813274-7c9036bd9a1f?w=3840&fit=crop 1x"
      style={
        {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": undefined,
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
  </noscript>
</span>
`;

exports[`supports layout="fill" 1`] = `
<span
  style={
    {
      "background": "none",
      "border": 0,
      "bottom": 0,
      "boxSizing": "border-box",
      "display": "block",
      "height": "initial",
      "left": 0,
      "margin": 0,
      "opacity": 1,
      "overflow": "hidden",
      "padding": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "width": "initial",
    }
  }
>
  <img
    alt="Feugiat sed lectus vestibulum mattis ullamcorper velit sed ullamcorper morbi tincidunt"
    data-nimg="fill"
    decoding="async"
    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
    style={
      {
        "border": "none",
        "bottom": 0,
        "boxSizing": "border-box",
        "display": "block",
        "height": 0,
        "left": 0,
        "margin": "auto",
        "maxHeight": "100%",
        "maxWidth": "100%",
        "minHeight": "100%",
        "minWidth": "100%",
        "objectFit": undefined,
        "objectPosition": undefined,
        "padding": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 0,
      }
    }
  />
  <noscript>
    <img
      alt="Feugiat sed lectus vestibulum mattis ullamcorper velit sed ullamcorper morbi tincidunt"
      data-nimg="fill"
      decoding="async"
      loading="lazy"
      sizes="100vw"
      src="https://images.unsplash.com/photo-1441974231531-c6227db76b6e?w=3840&fit=crop"
      srcSet="https://images.unsplash.com/photo-1441974231531-c6227db76b6e?w=640&fit=crop 640w, https://images.unsplash.com/photo-1441974231531-c6227db76b6e?w=750&fit=crop 750w, https://images.unsplash.com/photo-1441974231531-c6227db76b6e?w=828&fit=crop 828w, https://images.unsplash.com/photo-1441974231531-c6227db76b6e?w=1080&fit=crop 1080w, https://images.unsplash.com/photo-1441974231531-c6227db76b6e?w=1200&fit=crop 1200w, https://images.unsplash.com/photo-1441974231531-c6227db76b6e?w=1920&fit=crop 1920w, https://images.unsplash.com/photo-1441974231531-c6227db76b6e?w=2048&fit=crop 2048w, https://images.unsplash.com/photo-1441974231531-c6227db76b6e?w=3840&fit=crop 3840w"
      style={
        {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": undefined,
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
  </noscript>
</span>
`;

exports[`supports layout="fixed" 1`] = `
<span
  style={
    {
      "background": "none",
      "border": 0,
      "boxSizing": "border-box",
      "display": "inline-block",
      "height": 3036,
      "margin": 0,
      "opacity": 1,
      "overflow": "hidden",
      "padding": 0,
      "position": "relative",
      "width": 4554,
    }
  }
>
  <img
    alt="Tempor orci eu lobortis elementum nibh tellus molestie nunc non blandit massa"
    data-nimg="fixed"
    decoding="async"
    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
    style={
      {
        "border": "none",
        "bottom": 0,
        "boxSizing": "border-box",
        "display": "block",
        "height": 0,
        "left": 0,
        "margin": "auto",
        "maxHeight": "100%",
        "maxWidth": "100%",
        "minHeight": "100%",
        "minWidth": "100%",
        "objectFit": undefined,
        "objectPosition": undefined,
        "padding": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 0,
      }
    }
  />
  <noscript>
    <img
      alt="Tempor orci eu lobortis elementum nibh tellus molestie nunc non blandit massa"
      data-nimg="fixed"
      decoding="async"
      loading="lazy"
      src="https://images.unsplash.com/photo-1470770903676-69b98201ea1c?w=3840&fit=crop"
      srcSet="https://images.unsplash.com/photo-1470770903676-69b98201ea1c?w=3840&fit=crop 1x"
      style={
        {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": undefined,
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
  </noscript>
</span>
`;

exports[`supports layout="intrinsic" 1`] = `
<span
  style={
    {
      "background": "none",
      "border": 0,
      "boxSizing": "border-box",
      "display": "inline-block",
      "height": "initial",
      "margin": 0,
      "maxWidth": "100%",
      "opacity": 1,
      "overflow": "hidden",
      "padding": 0,
      "position": "relative",
      "width": "initial",
    }
  }
>
  <span
    style={
      {
        "background": "none",
        "border": 0,
        "boxSizing": "border-box",
        "display": "block",
        "height": "initial",
        "margin": 0,
        "maxWidth": "100%",
        "opacity": 1,
        "padding": 0,
        "width": "initial",
      }
    }
  >
    <img
      alt=""
      aria-hidden={true}
      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%274240%27%20height=%272832%27/%3e"
      style={
        {
          "background": "none",
          "border": 0,
          "display": "block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "padding": 0,
          "width": "initial",
        }
      }
    />
  </span>
  <img
    alt="Sed vulputate mi sit amet"
    data-nimg="intrinsic"
    decoding="async"
    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
    style={
      {
        "border": "none",
        "bottom": 0,
        "boxSizing": "border-box",
        "display": "block",
        "height": 0,
        "left": 0,
        "margin": "auto",
        "maxHeight": "100%",
        "maxWidth": "100%",
        "minHeight": "100%",
        "minWidth": "100%",
        "objectFit": undefined,
        "objectPosition": undefined,
        "padding": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 0,
      }
    }
  />
  <noscript>
    <img
      alt="Sed vulputate mi sit amet"
      data-nimg="intrinsic"
      decoding="async"
      loading="lazy"
      src="https://images.unsplash.com/photo-1604537466608-109fa2f16c3b?w=3840&fit=crop"
      srcSet="https://images.unsplash.com/photo-1604537466608-109fa2f16c3b?w=3840&fit=crop 1x"
      style={
        {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": undefined,
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
  </noscript>
</span>
`;

exports[`supports layout="responsive" 1`] = `
<span
  style={
    {
      "background": "none",
      "border": 0,
      "boxSizing": "border-box",
      "display": "block",
      "height": "initial",
      "margin": 0,
      "opacity": 1,
      "overflow": "hidden",
      "padding": 0,
      "position": "relative",
      "width": "initial",
    }
  }
>
  <span
    style={
      {
        "background": "none",
        "border": 0,
        "boxSizing": "border-box",
        "display": "block",
        "height": "initial",
        "margin": 0,
        "opacity": 1,
        "padding": 0,
        "paddingTop": "66.66666666666666%",
        "width": "initial",
      }
    }
  />
  <img
    alt="Sagittis vitae et leo duis ut diam quam nulla porttitor massa id neque"
    data-nimg="responsive"
    decoding="async"
    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
    style={
      {
        "border": "none",
        "bottom": 0,
        "boxSizing": "border-box",
        "display": "block",
        "height": 0,
        "left": 0,
        "margin": "auto",
        "maxHeight": "100%",
        "maxWidth": "100%",
        "minHeight": "100%",
        "minWidth": "100%",
        "objectFit": undefined,
        "objectPosition": undefined,
        "padding": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 0,
      }
    }
  />
  <noscript>
    <img
      alt="Sagittis vitae et leo duis ut diam quam nulla porttitor massa id neque"
      data-nimg="responsive"
      decoding="async"
      loading="lazy"
      sizes="100vw"
      src="https://images.unsplash.com/photo-1431794062232-2a99a5431c6c?w=3840&fit=crop"
      srcSet="https://images.unsplash.com/photo-1431794062232-2a99a5431c6c?w=640&fit=crop 640w, https://images.unsplash.com/photo-1431794062232-2a99a5431c6c?w=750&fit=crop 750w, https://images.unsplash.com/photo-1431794062232-2a99a5431c6c?w=828&fit=crop 828w, https://images.unsplash.com/photo-1431794062232-2a99a5431c6c?w=1080&fit=crop 1080w, https://images.unsplash.com/photo-1431794062232-2a99a5431c6c?w=1200&fit=crop 1200w, https://images.unsplash.com/photo-1431794062232-2a99a5431c6c?w=1920&fit=crop 1920w, https://images.unsplash.com/photo-1431794062232-2a99a5431c6c?w=2048&fit=crop 2048w, https://images.unsplash.com/photo-1431794062232-2a99a5431c6c?w=3840&fit=crop 3840w"
      style={
        {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": undefined,
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
  </noscript>
</span>
`;
